matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/ThirdPidApi.java

```diff
@@ -15,57 +16,100 @@
 
     /**
      * Request an email validation
+     *
      * @param clientSecret the client secret
-     * @param email the email address
-     * @param sendAttempt the attempt count
-     * @param nextLink the next link
-     * @param callback the asynchronous callback called with the response
+     * @param email        the email address
+     * @param sendAttempt  the attempt count
+     * @param nextLink     the next link
+     * @param callback     the asynchronous callback called with the response
      */
     @POST("/validate/email/requestToken")
-    void requestEmailValidation(@Query("client_secret") String clientSecret, @Query("email") String email, @Query("send_attempt") Integer sendAttempt, @Query("next_link") String nextLink, Callback<RequestEmailValidationResponse> callback);
+    void requestEmailValidation(@Query("client_secret") String clientSecret,
+                                @Query("email") String email,
+                                @Query("send_attempt") Integer sendAttempt,
+                                @Query("next_link") String nextLink, Callback<RequestEmailValidationResponse> callback);
+
+    /**
+     * Request a phone number validation
+     *
+     * @param clientSecret the client secret
+     * @param phoneNumber  the phone number
+     * @param country      the country
+     * @param sendAttempt  the attempt count
+     * @param nextLink     the next link
+     * @param callback     the asynchronous callback called with the response
+     */
+    @POST("/validate/msisdn/requestToken")
+    void requestPhoneNumberValidation(@Query("client_secret") String clientSecret,
+                                      @Query("phone_number") String phoneNumber,
+                                      @Query("country") String country,
+                                      @Query("send_attempt") Integer sendAttempt,
+                                      @Query("next_link") String nextLink,
+                                      Callback<RequestPhoneNumberValidationResponse> callback);
 
     /**
      * Request the ownership validation of an email address previously set
      * by {@link #requestEmailValidation(String, String, Integer, String, Callback)}.
-     * @param token the token generated by the requestToken call
+     *
+     * @param token        the token generated by the requestToken call
      * @param clientSecret the client secret which was supplied in the requestToken call
-     * @param sid the sid for the session
-     * @param callback asynchronous callback response
+     * @param sid          the sid for the session
+     * @param callback     asynchronous callback response
      */
     @POST("/validate/email/submitToken")
-    void requestEmailOwnershipValidation(@Query("token") String token, @Query("client_secret") String clientSecret, @Query("sid") String sid, Callback<Map<String,Object>> callback);
+    void requestEmailOwnershipValidation(@Query("token") String token,
```

Comment:
the both msisdn and email request scould be done in a single
@POST("/validate/{medium}/submitToken")
void requestOwnershipValidation(@Path("medium") String medium ...